handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -212,7 +212,46 @@ SSLEngineResult unwrap(SslHandler handler, ByteBuf in, int readerIndex, int len,
 
             @Override
             int calculateOutNetBufSize(SslHandler handler, int pendingBytes) {
-                return ReferenceCountedOpenSslEngine.calculateOutNetBufSize(pendingBytes);
+                return SslUtils.calculateOutNetBufSize(pendingBytes);
+            }
+        },
+        CONSCRYPT(true, COMPOSITE_CUMULATOR) {
+            @Override
+            SSLEngineResult unwrap(SslHandler handler, ByteBuf in,
+                                   int readerIndex, int len,
+                                   ByteBuf out) throws SSLException {
+                int nioBufferCount = in.nioBufferCount();
+                int writerIndex = out.writerIndex();
+                final SSLEngineResult result;
+                if (nioBufferCount > 1) {
+                    /**
+                     * If {@link org.conscrypt.OpenSslEngine} is in use,
+                     * we can use a special
+                     * {@link org.conscrypt.OpenSslEngine#unwrap(ByteBuffer[], ByteBuffer[])} method
+                     * that accepts multiple {@link ByteBuffer}s without additional memory copies.
+                     */
+                    org.conscrypt.OpenSSLEngineImpl conscryptEngine =
+                            ((ConscryptAlpnSslEngine) handler.engine).getWrappedConscryptEngine();
+                    try {
+                        handler.singleBuffer[0] = toByteBuffer(out, writerIndex,
+                                out.writableBytes());
+                        result = conscryptEngine
+                                .unwrap(in.nioBuffers(readerIndex, len), handler.singleBuffer);
+                        out.writerIndex(writerIndex + result.bytesProduced());
```

Comment:
I think this can be removed as it is done below (I already removed it for `TCNATIVE` [1])

[1] https://github.com/netty/netty/commit/2cff91804426399ac1293ed5056b1ecb2e87981a#diff-2fe7b22a8d650f1ea0bf56a809c061f9L201